### PR TITLE
Bounding box

### PR DESCRIPTION
**Pull request by TrevorDev**
https://github.com/BabylonJS/Babylon.js/issues/4310

Testing PG:
http://localhost:1338/Playground/index-local.html#7KX2R8#84

VR optimizations coming next.

### CODE REVIEW

File: src/Behaviors/Mesh/babylon.pointerDragBehavior.ts [hunk 1]
```diff
@@ -9,11 +9,18 @@ module BABYLON {
         private _pointerObserver:Nullable<Observer<PointerInfo>>;
         private static _planeScene:Scene;
         private _draggingID = -1;
+        // Debug mode will display drag planes to help visualize behavior
+        private _debugMode = false;
+        private _maxDragAngle = Math.PI/5;
         
         /**
          *  Fires each time the attached mesh is dragged with the pointer
+         *  delta between last drag position and current drag position in world space
```
Comment (deltakosh) [inline src/Behaviors/Mesh/babylon.pointerDragBehavior.ts]: needs to add "." Else the doc will just merge everything into one line

File: src/Gizmos/babylon.boundingBoxGizmo.ts [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+                    }
+                }
+            }
+        }
+
+        /**
+         * Disposes of the gizmo
+         */
+        public dispose(){
+            this.gizmoLayer.originalScene.onBeforeRenderObservable.remove(this._renderObserver);
```
Comment (deltakosh) [inline src/Gizmos/babylon.boundingBoxGizmo.ts]: what about the lines?

Comment (TrevorDev) [reply src/Gizmos/babylon.boundingBoxGizmo.ts]: The lines are disposed with the line below: this._lineBoundingBox.dispose();
or do you mean something else?

Comment (deltakosh) [reply src/Gizmos/babylon.boundingBoxGizmo.ts]: oh ok..did not see the addChild call
Cool!

**Pull request merged**